### DISCUSSION TITLE
Add an AUTHORS file?

### DISCUSSION THREAD

**Discussion by leighmcculloch** [vlang/v#10025] [Ideas]
Would the maintainers be up for adding an `AUTHORS` file and changing the license to reflect that copyright is held by the authors of the project?

This project's contributor base has grown in size significantly. According to GitHub the v repository has 416 contributors, which is 🎉. Projects of similar sizes often have an `AUTHORS` file that lists all contributors, and then the license references `The [Project Name]'s Authors` as the copyright holder.

Google has a good page that discusses the how and why of using an AUTHORS file. As noted in Google's doc, contributors hold copyright on their contributions regardless of whether an AUTHORS file exists, but it serves to make that more explicit.
https://opensource.google/docs/releasing/authors/

Examples of other similar projects that do this:
https://github.com/nodejs/node/blob/master/LICENSE#L4
https://github.com/nodejs/node/blob/master/AUTHORS
https://github.com/golang/go/blob/master/AUTHORS
https://github.com/golang/go/blob/master/LICENSE#L1
https://github.com/denoland/deno/blob/main/LICENSE.md

---

Generating an `AUTHORS` file is relatively simple and doesn't come with much of a technical burden. It can be generated from time to time with git, or can run as a GitHub Action either on PRs, or on merges and automatically push a change itself.

For example, the following commands will generate author files using git and unix tools:
```sh
git log --format='%aN <%aE>' | sort -uf
```
```sh
git shortlog -sne
```

Or, for example, the following short v script would generate similar:
```v
import os

pub fn main() {
	r := os.execute("git log --reverse --format='%aN,%aE'")
	if r.exit_code != 0 {
		eprintln("error: code $r.exit_code: $r.output")
	}

	mut authors := map[string][]string{}
	lines := r.output.split_into_lines()
	for l in lines {
		parts := l.split(',')
		name := parts[0]
		email := parts[1]
		mut emails := authors[name]
		if !(email in emails) {
			emails << email
		}
		authors[name] = emails
	}

	for author, emails in authors {
		println('$author ${emails.map('<$it>').join(' ')}')
	}
}
```

**Comment by bstnbuck** [comment]
Good idea!
As you said, this can also be done with GitHub actions. In the Marketplace there are already ready-made tools that could be useful. Something like this is nice:
https://github.com/marketplace/actions/contribute-list

**Comment by JalonSolov** [comment]
Or you could just click on "Contributors" on the main repo Github page.  Over 400 names...

**Comment by fasakinhenry** [comment]
The `AUTHORS` file serves as a file to list all the contributors in a particular project (both open and closed-sourced projects). You can easily create a file name `AUTHORS` in the top level of your repository and add all the names of the contributors. It really helps to give tribute to the project contributors as they have the right to the ownership of their intellectual input. This is a basic format of the `AUTHORS` file.

```
# This is the list of [Project Name]'s contributors.

Henry Fasakin
Fasakin Sola
```

don't mind me I used my surname in both names. Seems I have turned the project into a family asset🤣😭😂

**Reply by fasakinhenry** [reply]
I don't know but it will be nice to write their names in front like this

```
# This is the list of [Project Name]'s contributors.

Henry Fasakin <example@gmail.com>
Fasakin Sola <example2@gmail.com>
```

**Comment by Wajinn** [comment]
While it might seem like a good idea on the surface, looks like it has the potential to stir up unnecessary drama. A big issue that I see, with a list of names, is how much of a contribution that was made would be unknown. A person could have contributed only once and has 1 commit, but could be put on the list of names with someone with 500 or more commits.

As JalonSolov has mentioned, there is a contributors link, and it has detailed information on commit history and not just a name. Of the opinion that would be a better way, which is to link to the contributors and possibly [vlang members](https://github.com/orgs/vlang/people). Lastly, there shouldn't be any confusion, as the project is under the MIT license and that governs the contributions made to it.
